src/cryptoadvance/specter/global_search.py

```diff
@@ -0,0 +1,260 @@
+            ],
+        )
+        addresses_change = HtmlElement(
+            addresses,
+            id=(
+                f"addresses-table-{wallet.alias}",
+                "shadowRoot",
+                "change-addresses-view-btn",
+            ),
+            function=lambda x: search_in_structure(
+                x, wallet.addresses_info(is_change=True)
+            ),
+            visible_on_endpoints=[
+                url_for("wallets_endpoint.wallet", wallet_alias=wallet.alias)
+            ],
+        )
+
+        recieve = HtmlElement(
+            sidebar_wallet,
+            id="btn_receive",
+            function=lambda x: search_in_structure(x, [wallet.address]),
+            visible_on_endpoints=[
+                url_for("wallets_endpoint.wallet", wallet_alias=wallet.alias)
+            ],
+        )
+
+        send = HtmlElement(
+            sidebar_wallet,
+            id="btn_send",
+            visible_on_endpoints=[
+                url_for("wallets_endpoint.wallet", wallet_alias=wallet.alias)
+            ],
+        )
+        unsigned = HtmlElement(
+            send,
+            id="btn_send_pending",
+            function=lambda x: search_in_structure(
+                x, [psbt.to_dict() for psbt in wallet.pending_psbts.values()]
+            ),
+            visible_on_endpoints=[
+                url_for("wallets_endpoint.wallet", wallet_alias=wallet.alias)
+            ],
+        )
+
+    def add_all_in_devices(device):
+        sidebar_device = HtmlElement(devices, id=f"device_list_item_{device.alias}")
+        device_names = HtmlElement(
+            sidebar_device,
+            id="title",
+            function=lambda x: search_in_structure(x, [device.alias]),
+            visible_on_endpoints=[
+                url_for("devices_endpoint.device", device_alias=device.alias)
+            ],
+        )
+        device_keys = HtmlElement(
+            sidebar_device,
+            id="keys-table-header-key",
+            function=lambda x: search_in_structure(x, [key for key in device.keys]),
+            visible_on_endpoints=[
+                url_for("devices_endpoint.device", device_alias=device.alias)
+            ],
+        )
+
+    for wallet in specter.wallet_manager.wallets.values():
+        add_all_in_wallet(wallet)
+    for device in specter.device_manager.devices.values():
+        add_all_in_devices(device)
+    return html_root
+
+
+def apply_search_on_dict(search_term, html_root):
+    "Given an html_root it will call the child.function for all childs that do not have any children"
+    end_nodes = html_root.calculate_end_nodes()
+    for end_node in end_nodes:
+        end_node.result = end_node.function(search_term)
+    return html_root
+
+
+def do_global_search(search_term, specter):
+    "Builds the HTML Tree if ncessary (only do it once) and then calls the functions in it to search for the search_term"
+    global HTML_ROOT
```

Comment:
I'm not sure that's a good idea?! Is that possible with multi-user? How expensive is the build-up of the tree? Maybe we can do it every time?

Reply:
Yes, you are right, I didn't think about multi-user yet.  Please do not put in too much effort now to review the code. Before I put more work in the code (and have to change a lot), I am seeking feedback, if this  feature is wanted, and how the UI should look like.